src/simpleNbox.cpp

```diff
@@ -143,16 +184,34 @@ void SimpleNbox::setData( const std::string &varName,
             set_c0(data.getUnitval(U_PPMV_CO2).value(U_PPMV_CO2));
         }
         else if( varNameParsed == D_VEGC ) {
-            H_ASSERT( data.date == Core::undefinedIndex(), "date not allowed" );
+            // For `veg_c`, `detritus_c`, and `soil_c`, if date is not
+            // provided, set only the "current" model pool, without
+            // touching the time series variable. This is to
+            // accommodate the way the INI file is parsed. For
+            // interactive use, you will usually want to pass the date
+            // -- otherwise, the current value will be overridden by a
+            // `reset` (which includes code like `veg_c = veg_c_tv.get(t)`).
+            //
+            // TODO: Should we create special variables `veg_c0`,
```

Comment:
Converted to an issue in case we want to discuss further: https://github.com/JGCRI/hector/issues/318

Removed the comment in 10d3ac2f353bc946bcfc5560802aff6a67f9f9e6.